Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,100 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows { get; }
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            List<string> states = CsvRows.
+                Select(person => person.Split(',')[6]).
+                Distinct().
+                OrderBy(state => state).
+                ToList();
+
+            return states;
+            //ask preference on where the . belongs on the above. end of row or beginning.
+        }
 
         // 3.
         public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+        {
+            IEnumerable<string> states = GetUniqueSortedListOfStatesGivenCsvRows();
+            states.ToArray();
+            return string.Join(",", states);
+        }
 
         // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+        public IEnumerable<IPerson> People { get; private set; }
 
         // 5.
-        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(
-            Predicate<string> filter) => throw new NotImplementedException();
+        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(Predicate<string> filter)
+        {
+            List<IPerson> persons = People.ToList();
+            IEnumerable<(string FirstName, string LastName)> results = People.ToList().
+                FindAll(person => filter(person.EmailAddress)).
+                Select(person => (person.FirstName, person.LastName));

```

Comment:
You don't need to convert this to a list if you use LINQ operations (`FindAll` is a list operation, hence why it only works on a list object)

Consider the following solution:
```suggestion
            IEnumerable<(string FirstName, string LastName)> results = People
                .Where(person => filter(person.EmailAddress))
                .Select(person => (person.FirstName, person.LastName));
```